packages/mana/consensusbasevector.go

```diff
@@ -36,6 +36,66 @@ func (c *ConsensusBaseManaVector) Has(nodeID identity.ID) bool {
 	return exists
 }
 
+// BuildPastBaseVector builds a consensus base mana vector from past events upto time `t`
+func (c *ConsensusBaseManaVector) BuildPastBaseVector(eventsLog []Event, t time.Time) error {
+	emptyID := identity.ID{}
+	if len(c.vector) == 0 {
+		c.vector = make(map[identity.ID]*ConsensusBaseMana)
```

Comment:
a freshly created vector (with `NewBaseManaVector()`) has length 0, but the map is initialized. Check for `nil` instead.